File: articles/three-good-things-exercise/go/appreciative_awareness_index.go
package main

import "fmt"

type ThreeGoodThingsIndicators struct {
	LifeSatisfaction      float64
	DepressiveSymptoms   float64
	GratitudeScore       float64
	PositiveEventSalience float64
	PerceivedSupport     float64
	ReflectionDepth      float64
	StressLoad           float64
	Acceptability        float64
	ContextFit           float64
}

func AppreciativeAwarenessIndex(x ThreeGoodThingsIndicators) float64 {
	return (x.GratitudeScore +
		x.PositiveEventSalience +
		x.PerceivedSupport +
		x.ReflectionDepth +
		x.Acceptability +
		x.ContextFit -
		x.StressLoad) / 7.0
}

func NetWellbeingIndex(x ThreeGoodThingsIndicators) float64 {
	return x.LifeSatisfaction +
		x.GratitudeScore +
		x.PositiveEventSalience +
		x.PerceivedSupport +
		x.ReflectionDepth -
		x.DepressiveSymptoms -
		x.StressLoad
}

func main() {
	example := ThreeGoodThingsIndicators{7.4, 3.6, 7.2, 7.1, 7.3, 7.1, 3.2, 7.7, 7.6}
	fmt.Printf("Appreciative awareness index: %.3f\n", AppreciativeAwarenessIndex(example))
	fmt.Printf("Net wellbeing index: %.3f\n", NetWellbeingIndex(example))
}

File: articles/three-good-things-exercise/rust/appreciative_awareness_index.rs
struct ThreeGoodThingsIndicators {
    life_satisfaction: f64,
    depressive_symptoms: f64,
    gratitude_score: f64,
    positive_event_salience: f64,
    perceived_support: f64,
    reflection_depth: f64,
    stress_load: f64,
    acceptability: f64,
    context_fit: f64,
}

fn appreciative_awareness_index(x: &ThreeGoodThingsIndicators) -> f64 {
    (x.gratitude_score
        + x.positive_event_salience
        + x.perceived_support
        + x.reflection_depth
        + x.acceptability
        + x.context_fit
        - x.stress_load)
        / 7.0
}

fn net_wellbeing_index(x: &ThreeGoodThingsIndicators) -> f64 {
    x.life_satisfaction
        + x.gratitude_score
        + x.positive_event_salience
        + x.perceived_support
        + x.reflection_depth
        - x.depressive_symptoms
        - x.stress_load
}

fn main() {
    let example = ThreeGoodThingsIndicators {
        life_satisfaction: 7.4,
        depressive_symptoms: 3.6,
        gratitude_score: 7.2,
        positive_event_salience: 7.1,
        perceived_support: 7.3,
        reflection_depth: 7.1,
        stress_load: 3.2,
        acceptability: 7.7,
        context_fit: 7.6,
    };

    println!("Appreciative awareness index: {:.3}", appreciative_awareness_index(&example));
    println!("Net wellbeing index: {:.3}", net_wellbeing_index(&example));
}
